dotCMS/src/integration-test/java/com/dotcms/contenttype/business/DotAssetBaseTypeToContentTypeStrategyImplTest.java

```diff
@@ -0,0 +1,118 @@
+import com.dotcms.util.CollectionsUtils;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.portlets.folders.business.FolderAPI;
+import com.dotmarketing.util.FileUtil;
+import com.dotmarketing.util.UUIDGenerator;
+import com.liferay.portal.model.User;
+import com.liferay.portal.util.PortalUtil;
+import com.liferay.portal.util.WebKeys;
+import org.glassfish.jersey.internal.util.Base64;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+
+public class DotAssetBaseTypeToContentTypeStrategyImplTest  extends IntegrationTestBase {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    /**
+     * 1) creates a dotAsset content type for text plain
+     * 2) creates a temporal file asset and gets the id
+     * 3) creates a map with the contentlet properties (including as a file asset the temporal file id)
+     * 4) runs the BaseTypeToContentTypeStrategy in order to get the new content type using the temporal file
+     * @throws Exception
+     */
+    @Test
+    public void test_apply_with_temporal_file() throws Exception {
+
+        // creates a dotAsset for text files
+        final String variable = "testDotAsset" + System.currentTimeMillis();
+        final ContentType dotAssetContentType = APILocator.getContentTypeAPI(APILocator.systemUser()).
+                save(ContentTypeBuilder.builder(DotAssetContentType.class).folder(
+                FolderAPI.SYSTEM_FOLDER).host(Host.SYSTEM_HOST).name(variable)
+                .owner(APILocator.systemUser().getUserId()).build());
+
+        final Map<String, Field> fieldMap = dotAssetContentType.fieldMap();
+        com.dotcms.contenttype.model.field.Field binaryField           = fieldMap.get(DotAssetContentType.ASSET_FIELD_VAR);
+        final FieldVariable allowFileTypes = ImmutableFieldVariable.builder().key(BinaryField.ALLOWED_FILE_TYPES)
+                .value("text/plain").fieldId(binaryField.id()).build();
+        binaryField.constructFieldVariables(Arrays.asList(allowFileTypes));
+
+        APILocator.getContentTypeAPI(APILocator.systemUser()).save(dotAssetContentType);
+        APILocator.getContentTypeFieldAPI().save(binaryField, APILocator.systemUser());
+        APILocator.getContentTypeFieldAPI().save(allowFileTypes, APILocator.systemUser());
+
+        final Optional<BaseTypeToContentTypeStrategy> baseTypeToContentTypeStrategy =
+                BaseTypeToContentTypeStrategyResolver.getInstance().get(BaseContentType.DOTASSET);
+
+        Assert.assertTrue(baseTypeToContentTypeStrategy.isPresent());
+        MockHeaderRequest request = new MockHeaderRequest(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'request' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093431)